lxd/seccomp.go

```diff
@@ -441,6 +443,41 @@ func NewSeccompServer(d *Daemon, path string) (*SeccompServer, error) {
 	return &s, nil
 }
 
+func doMknod(dev types.Device, msg *C.struct_seccomp_notify_proxy_msg) error {
+	errnoMsg, err := shared.RunCommand(util.GetExecPath(),
+		"forkmknod", dev["pid"], dev["path"],
+		dev["mode_t"], dev["dev_t"])
+	if err != nil {
+		goErrno, err2 := strconv.Atoi(errnoMsg)
+		if err2 == nil {
+			C.seccomp_notify_mknod_update_response(msg, -C.int(goErrno))
+		} else {
+			C.seccomp_notify_mknod_update_response(msg, -C.EPERM)
+		}
+	} else {
+		C.seccomp_notify_mknod_update_response(msg, 0)
+	}
+
+	return err
+}
+
+func doInjectDevice(dev types.Device, request_pid int, msg *C.struct_seccomp_notify_proxy_msg, d *Daemon) error {
+	c, err := findContainerForPid(int32(msg.monitor_pid), d)
+	if err == nil {
+		err = c.InsertSeccompUnixDevice("forkmknod.unix", dev, request_pid)
```

Comment:
shouldn't we use the task's pid or something in there for when we eventually allow handling of multiple parallel requests for the same container?

Reply:
Not sure what exactly you're referring to here but request pid is the tasks pid.

Reply:
In the file name

Reply:
*shrug*, sure